src/MyTested.AspNetCore.Mvc.Caching/Builders/Data/MemoryCacheWithBuilder.cs

```diff
@@ -4,45 +4,33 @@
     using System.Collections.Generic;
     using Contracts.Data;
     using Microsoft.Extensions.Caching.Memory;
-    using Microsoft.Extensions.DependencyInjection;
     using Utilities.Extensions;
 
-    /// <summary>
-    /// Used for building mocked <see cref="IMemoryCache"/>.
-    /// </summary>
-    public class MemoryCacheBuilder : IAndMemoryCacheBuilder
+    /// <inheritdoc />
+    public class MemoryCacheWithBuilder : MemoryCacheBaseBuilder, IAndWithMemoryCacheBuilder
```

Comment:
The class and the interface should have similar names. Either the class should be `WithMemoryCacheBuilder` or the interface should be `IMemoryCacheWithBuilder`. I prefer the `WithMemoryCacheBuilder` name.